odl/discr/discr_ops.py

```diff
@@ -101,56 +111,353 @@ def _call(self, x, out=None):
+        ----------
+        domain : uniform `DiscreteLp`
+            Uniformly discretized space, the operator can be applied
+            to its elements.
+        range : uniform `DiscreteLp`, optional
+            Uniformly discretized space in which the result of the
+            application of this operator lies.
+            For the default ``None``, a space with the same attributes
+            as ``domain`` is used, except for its shape, which is set
+            to ``ran_shp``.
+        ran_shp : sequence of int
+            Shape of the range of this operator. This can be provided
+            instead of ``range`` and is mandatory if ``range`` is
+            ``None``.
+        offset : int or sequence of int, optional
+            Number of cells to add to/remove from the left of
+            ``domain.partition``. By default, the difference is
+            distributed evenly, with preference for left in case of
+            ambiguity.
+            This option is can only be used together with ``ran_shp``.
+        pad_mode : str, optional
+            Method to be used to fill in missing values in an enlarged array.
+
+            ``'constant'``: Fill with zeros.
+
+            ``'symmetric'``: Reflect at the boundaries, not doubling the
+            outmost values. This requires left and right padding sizes
+            to be strictly smaller than the original array shape.
+
+            ``'periodic'``: Fill in values from the other side, keeping
+            the order. This requires left and right padding sizes to be
+            at most as large as the original array shape.
+
+            ``'order0'``: Extend constantly with the outmost values
+            (ensures continuity).
+
+            ``'order1'``: Extend with constant slope (ensures continuity of
+            the first derivative). This requires at least 2 values along
+            each axis where padding is applied.
+
+        discr_kwargs: dict, optional
+            Keyword arguments passed to the `uniform_discr` constructor.
+        """
+        # Catch pad_const so it's not silently ignored
+        if 'pad_const' in kwargs:
+            raise TypeError("got invalid argument 'pad_const'")
+        super().__init__(domain, range, ran_shp, **kwargs)
+
+    def _call(self, x, out):
+        """Implement ``self(x, out)``."""
+        # TODO: simplify once context manager is available
+        out[:] = resize_array(x.asarray(), self.range.shape,
+                              offset=self.offset, pad_mode=self.pad_mode,
+                              pad_const=0, direction='adjoint',
+                              out=out.asarray())
+
+    @property
+    def adjoint(self):
+        """Adjoint of this operator."""
+        return ResizingOperator(domain=self.range, range=self.domain,
+                                pad_mode=self.pad_mode, pad_const=0.0)
+
+
+def _resize_discr(discr, newshp, offset, discr_kwargs):
+    """Return ``discr`` resized to ``newshp``.
```

Comment:
I disagree with this docstring, it barely retents `discr`'s properties at all, basically only taking the underlying domain and size. Things like `dspace`, `dtype` etc seem to be taken from `discr_kwargs`.